### PR TITLE
Backend improvements

### PR DESCRIPTION
**Pull request by lzap**
This is a patch that improves backend services to the degree that I am able to use compose for my own development setup for backend. There are several commits:

Commit: Migration was merged from a separate container into the main container and is performed before application start. This is similar to what sources does. Since migration is extremely fast (takes about a second), running it before running the service should not be a problem.

Commit: Copied and updated the seed script and configuration for sources. This needs to be done manually, not from container, as sources are now exposed. This takes seed script from our main repo here, I verified it works.

```
[lzap@zzzap provisioning-compose]$ ./sources_seed/seed.sh 
{"sources":[{"availability_status":"in_progress","id":"1","created_at":"2023-07-20T12:33:47Z","updated_at":"2023-07-20T12:33:47Z","name":"Amazon source","uid":"2256035d-f6b4-423a-82ba-2461c0d4d43b","app_creation_workflow":"manual_configuration","source_type_id":"1"}],"applications":[{"id":"1","created_at":"2023-07-20T12:33:47Z","updated_at":"2023-07-20T12:33:47Z","availability_status":"in_progress","source_id":"1","application_type_id":"1"}],"endpoints":[],"authentications":[{"id":"1","authtype":"provisioning-arn","username":"arn:aws:iam::399777895069:role/satellite-services-role","availability_status":"in_progress","resource_type":"Application","resource_id":"1"}]}
{"sources":[{"availability_status":"in_progress","id":"2","created_at":"2023-07-20T12:33:47Z","updated_at":"2023-07-20T12:33:47Z","name":"Azure source","uid":"d7cc28a7-1f6b-443b-99d9-61c507b02058","app_creation_workflow":"manual_configuration","source_type_id":"8"}],"applications":[{"id":"2","created_at":"2023-07-20T12:33:47Z","updated_at":"2023-07-20T12:33:47Z","availability_status":"in_progress","source_id":"2","application_type_id":"1"}],"endpoints":[],"authentications":[{"id":"2","authtype":"provisioning_lighthouse_subscription_id","username":"2fd0879a-3149-4750-a10f-803f73586ad3","availability_status":"in_progress","resource_type":"Application","resource_id":"2"}]}
{"sources":[{"availability_status":"in_progress","id":"3","created_at":"2023-07-20T12:33:47Z","updated_at":"2023-07-20T12:33:47Z","name":"Google source","uid":"f1a04b15-99b6-4c3c-8b60-5d18dfb92d2c","app_creation_workflow":"manual_configuration","source_type_id":"12"}],"applications":[{"id":"3","created_at":"2023-07-20T12:33:47Z","updated_at":"2023-07-20T12:33:47Z","availability_status":"in_progress","source_id":"3","application_type_id":"1"}],"endpoints":[],"authentications":[{"id":"3","authtype":"provisioning_project_id","username":"redhat-360508","availability_status":"in_progress","resource_type":"Application","resource_id":"3"}]}
[lzap@zzzap provisioning-compose]$ curl-local GET /provisioning/v1/sources
[{"id":"1","name":"Amazon source","source_type_id":"1","uid":"2256035d-f6b4-423a-82ba-2461c0d4d43b"},{"id":"2","name":"Azure source","source_type_id":"8","uid":"d7cc28a7-1f6b-443b-99d9-61c507b02058"},{"id":"3","name":"Google source","source_type_id":"12","uid":"f1a04b15-99b6-4c3c-8b60-5d18dfb92d2c"}]
```

Commit: Allow overriding of exposed ports. For backend development, I need to have a lot of ports exposed (kafka, sources, ...) this change allows users to modify `compose.example.env` and change port numbers (8000 is not available on my machine. On top of that, data directory now can be changed too.

Commit: Kafka container does not use root account therefore in rootless podman the account is not properly mapped leading to permission errors. This commit introduces a `prepare-podman.sh` script which cleans the directory and sets up permission in a way that it will work with rootless podman.

### CODE REVIEW

File: compose.yml [hunk 1]
```diff
@@ -165,7 +158,7 @@ services:
     volumes:
       - ./frontend/:/frontend
     ports: 
-      - 1337:1337
+      - ${EXPOSED_FRONTEND_DEV_PORT:-8008}:${FRONTEND_DEV_PORT:-8000}
```
Comment (amirfefer) [inline compose.yml]: wouldn't it be better to keep the default chrome frontend as 1337?
also, 8000 is already the default for the backend

File: sources_seed/identity_header.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#
+une="$(uname -s)"
+case "${une}" in
+  Linux*)     B64OPTS="-w0";;
+  Darwin*)    B64OPTS="-b0";;
+  *)          echo "Unknown system ${une}" && exit 1
+esac
+
+if [ -n "${1}" ] && [ -n "${2}" ]; then
+  IDENTITY=$(echo "{\"identity\": {\"type\": \"User\", \"account_number\":\"$1\", \"internal\":{\"org_id\":\"$2\"}}}" | base64 $B64OPTS)
```
Comment (amirfefer) [inline sources_seed/identity_header.sh]: The notifications service  needs  the identity header to include a user object to function properly, as following
```json
"identity":{"type":"User","account_number":"","org_id":"","user":{"username":"","email":"","first_name":"","last_name":""}}
```
would we like to add it to this generator to make it easier to consume?

File: compose.yml [hunk 3]
```diff
@@ -106,16 +108,18 @@ services:
...
-      migrate:
-        condition: service_completed_successfully
       backend-db:
         condition: service_healthy
       init-kafka:
         condition: service_completed_successfully
+    entrypoint:
+      - bash
+      - "-c"
+      - "/pbackend migrate && /pbackend api"
```
Comment (amirfefer) [inline compose.yml:118]: this overrides the default entrypoint from the Dockerfile which enables live reload, 

```yml
entrypoint: 
      - bash
      -  "-c"
      - /pbackend migrate && $HOME/go/bin/CompileDaemon -build="go build -buildvcs=false -o /build/pbackend ./cmd/pbackend" -command="/build/pbackend api"
```

or instead, we can add an env variable for enabling live reload

**Review by amirfefer — COMMENTED** [review]
Thanks @lzap, this change works under docker as well 👍 
I left a few comments

File: compose.yml (hunk 1 above)
Comment (lzap) [reply compose.yml]: Absolutely this was a copy-paste error.

File: sources_seed/identity_header.sh (hunk 2 above)
Comment (lzap) [reply sources_seed/identity_header.sh]: Sure.

File: compose.yml (hunk 3 above)
Comment (lzap) [reply compose.yml:118]: This line is for backend not backend-dev, but you are right. This was not intentional.

**Comment by lzap** [comment]
Rebased.

File: compose.yml [hunk 4]
```diff
@@ -125,34 +129,23 @@ services:
...
         condition: service_healthy
       init-kafka:
         condition: service_completed_successfully
     build:
       context: ./backend
       dockerfile: ../Dockerfile.backend.dev
+    entrypoint:
+      - bash
+      - "-c"
+      - "go run ./cmd/pbackend migrate && $HOME/go/bin/CompileDaemon -build='go build -buildvcs=false -o /build/pbackend ./cmd/pbackend' -command='/build/pbackend api'"
```
Comment (amirfefer) [inline compose.yml]: the var `$HOME` results with the host's home, not the container's home, and it fails to launch.
we can change the `ENTRYPOINT` in the dockerfike instead, or using the hardcoded home of the container `/opt/app-root/src/`

**Review by amirfefer — CHANGES_REQUESTED** [review]
Thanks @lzap,
The backend-dev migrates well but fails to run the api server

**Comment by lzap** [comment]
Sure how about this (untested)? If it does not work I can hardcode the path.

**Review by amirfefer — APPROVED** [review]
Great, works well 👍

**Pull request merged**